renderdoc/driver/metal/metal_types.cpp

```diff
@@ -0,0 +1,50 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#include "metal_types.h"
+
+RDCCOMPILE_ASSERT(sizeof(NS::Integer) == sizeof(std::intptr_t), "NS::Integer size does not match");
+RDCCOMPILE_ASSERT(sizeof(NS::UInteger) == sizeof(std::uintptr_t),
+                  "NS::UInteger size does not match");
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, NS::String *&el)
+{
+  const char *cStr = el ? el->utf8String() : NULL;
+  SERIALISE_ELEMENT(cStr);
```

Comment:
Instead of using `SERIALISE_ELEMENT` which will treat NSString like a struct with a string member, you should forward the call on to `DoSerialise(ser, str)`. That will serialise "as-if" it were equivalent, so the NSString itself will be treated as just a string, not a struct.

See how interfaces are serialised 'as-if' they are a ResourceId.

Reply:
Thanks yeah that looks a lot nicer resource inspector display

Reply:
d62879c53e9e464f4ba383b323c9783e2e056eec